assembly/src/library/masl.rs

```diff
@@ -315,17 +305,25 @@ impl Serializable for MaslLibrary {
         // this assert is OK because maximum number of modules is enforced by Library constructor
         debug_assert!(modules.len() <= MAX_MODULES, "too many modules");
 
+        // write the flag whether the source locations should be serialized
+        target.write_bool(self.has_source_locations);
```

Comment:
Should we replace `has_source_locations` in `MaslLibrary` with full `serde_options`? This way, we treat both options (and potential future options consistently).